pkg/operators/process/process.go

```diff
@@ -0,0 +1,778 @@
+// %ESCAPE_SPACE: ('\f', '\n', '\r', '\t', '\v')
+// %ESCAPE_SPECIAL: ('\"', '\\', '\a', '\e')
+func unescapeCommand(cmd string) string {
+	// If there are no escape sequences, return the original string
+	if !strings.Contains(cmd, "\\") {
+		return cmd
+	}
+
+	var builder strings.Builder
+	builder.Grow(len(cmd)) // Pre-allocate capacity to avoid reallocations
+
+	for i := 0; i < len(cmd); i++ {
+		// Check for escape sequence
+		if cmd[i] == '\\' && i+1 < len(cmd) {
+			switch cmd[i+1] {
+			// Special whitespace characters
+			case 'f':
+				builder.WriteByte('\f')
+			case 'n':
+				builder.WriteByte('\n')
+			case 'r':
+				builder.WriteByte('\r')
+			case 't':
+				builder.WriteByte('\t')
+			case 'v':
+				builder.WriteByte('\v')
+			// Special characters
+			case '"':
+				builder.WriteByte('"')
+			case '\\':
+				builder.WriteByte('\\')
+			case 'a':
+				builder.WriteByte('\a')
+			case 'e':
+				builder.WriteByte(27) // ASCII 27 for escape character
+			default:
+				// Unknown escape sequence, keep both characters
+				builder.WriteByte('\\')
+				builder.WriteByte(cmd[i+1])
+			}
+			i++ // Skip the next character as we've already processed it
+		} else {
+			builder.WriteByte(cmd[i])
+		}
+	}
+
+	return builder.String()
+}
+
+// parseStatFile parses the /proc/[pid]/stat file, handling the comm field correctly
+// The comm field is enclosed in parentheses and may contain spaces
+func parseStatFile(statData string) ([]string, error) {
+	// The format of /proc/[pid]/stat is:
+	// pid (comm) state ppid ... and more fields
+	// The challenge is that comm can contain spaces and parentheses
+
+	// First, find the closing parenthesis for the comm field
+	commEndIdx := strings.LastIndex(statData, ")")
+	if commEndIdx == -1 {
+		return nil, fmt.Errorf("invalid stat file format: missing closing parenthesis in %q", statData)
+	}
+
+	// Find the opening parenthesis
+	commStartIdx := strings.Index(statData, "(")
+	if commStartIdx == -1 {
+		return nil, fmt.Errorf("invalid stat file format: missing opening parenthesis in %q", statData)
+	}
+
+	// Validate that the opening parenthesis comes before the closing one
+	if commStartIdx >= commEndIdx {
+		return nil, fmt.Errorf("invalid stat file format: parentheses in wrong order in %q", statData)
+	}
+
+	// Extract the pid (field 1)
+	pidStr := strings.TrimSpace(statData[:commStartIdx])
+
+	// Extract the comm (field 2)
+	comm := statData[commStartIdx+1 : commEndIdx]
+
+	// Extract the rest of the fields
+	// We only need a few specific fields, so we'll parse them directly
+	// without creating a full array of all fields
+	restData := statData[commEndIdx+1:]
+	fields := strings.Fields(restData)
+
+	// We need at least 21 fields after the comm field to access field 22
+	if len(fields) < 21 {
+		return nil, fmt.Errorf("invalid stat file format: not enough fields")
+	}
+
+	// Create a result array with only the fields we need
+	// This is more efficient than creating a full array with all fields
```

Comment:
Why is it more efficient? I would imagine the CPU usage to parse numbers is not too bad, but the dynamic allocations and the branches are more of concern.

Reply:
More efficient than filling out all of them. I think changing everything to use []byte instead of string should have already improved the situation.